### PR TITLE
Split out queued phase from QueryManager

### PR DESCRIPTION
**Pull request by dain**
Move queued phase of query from QueryManager to a new dispatcher service. This
change is in preparation for adding a optional new server that moves the queue
phase to a separate process.

### CODE REVIEW

File: presto-main/src/main/java/com/facebook/presto/execution/QueryStateTimer.java [hunk 1]
```diff
@@ -73,15 +75,27 @@ private void beginWaitingForResources(long now)
...
 
+    public void beginDispatching()
+    {
+        beginDispatching(tickerNanos());
+    }
+
+    private void beginDispatching(long now)
+    {
+        beginWaitingForResources(now);
+        resourceWaitingTime.compareAndSet(null, nanosSince(beginResourceWaitingNanos, now));
```
Comment (raghavsethi) [inline presto-main/src/main/java/com/facebook/presto/execution/QueryStateTimer.java]: I'm assuming resource waiting time is deliberately 0 for this commit.

**Comment by findepi** [comment]
> `Remove system startup minimum worker requirement`

is this removing a functionality or moving it elsewhere?

File: presto-main/src/main/java/com/facebook/presto/server/protocol/Query.java [hunk 2]
```diff
@@ -559,8 +499,9 @@ private synchronized URI createNextResultsUri(String scheme, UriInfo uriInfo)
     {
         return uriInfo.getBaseUriBuilder()
                 .scheme(scheme)
-                .replacePath("/v1/statement")
+                .replacePath("/v1/statement/executing")
```
Comment (raghavsethi) [inline presto-main/src/main/java/com/facebook/presto/server/protocol/Query.java]: I'm assuming you've talked to @martint or @electrum offline about the specific API change. I know we had a general discussion previously.

File: presto-tests/src/test/java/com/facebook/presto/execution/resourceGroups/db/TestQueuesDb.java [hunk 3]
```diff
@@ -250,9 +253,9 @@ public void testSelectorPriority()
         QueryId secondQuery = createQuery(queryRunner, dashboardSession(), LONG_LASTING_QUERY);
         waitForQueryState(queryRunner, secondQuery, FAILED);
 
-        resourceGroup = queryManager.getFullQueryInfo(secondQuery).getResourceGroupId();
-        assertTrue(resourceGroup.isPresent());
-        assertEquals(resourceGroup.get(), createResourceGroupId("global", "user-user", "reject-all-queries"));
+        DispatcherQueryManager dispatcherQueryManager = queryRunner.getCoordinator().getDispatcherQueryManager();
+        BasicQueryInfo basicQueryInfo = dispatcherQueryManager.getQueryInfo(secondQuery);
+        assertEquals(basicQueryInfo.getErrorCode(), QUERY_QUEUE_FULL.toErrorCode());
```
Comment (raghavsethi) [inline presto-tests/src/test/java/com/facebook/presto/execution/resourceGroups/db/TestQueuesDb.java]: Hmm this feels like a different test. Is there a reason why we won't also check the resource group?

File: presto-tests/src/test/java/com/facebook/presto/tests/TestMetadataManager.java [hunk 4]
```diff
@@ -113,20 +113,22 @@ public void testMetadataIsClearedAfterQueryFailed()
     public void testMetadataIsClearedAfterQueryCanceled()
             throws Exception
     {
-        QueryManager queryManager = queryRunner.getCoordinator().getQueryManager();
-        QueryId queryId = queryManager.createQueryId();
-        queryManager.createQuery(
+        DispatcherQueryManager dispatcherQueryManager = queryRunner.getCoordinator().getDispatcherQueryManager();
```
Comment (raghavsethi) [inline presto-tests/src/test/java/com/facebook/presto/tests/TestMetadataManager.java]: You're doing this again on L126

**Comment by tooptoop4** [comment]
@raghavsethi can this please be merged? There is a large security hole without this

**Comment by raghavsethi** [comment]
See https://github.com/prestodb/presto/pull/12801#issuecomment-513941778

**Pull request closed without merging**